### PR TITLE
okd 4.10.0-0.okd-2022-04-23-131357 rebase to ocp 4.10.16

### PR DESCRIPTION
**Pull request by sallyom**
rebase to`OCP 4.10.16` 
@fzdarsky @mangelajo

### CODE REVIEW

**Comment by sallyom** [comment]
@mangelajo the rpm install is failing due to  `DaemonSet:kube-system/kube-flannel-ds` fails. Is the `flannel-cni:4.10.13` image built w/in the PR CI?

**Comment by mangelajo** [comment]
@sallyom can you give it a try to updating release_amd64.go flannel-cni reference https://github.com/redhat-et/microshift/blob/main/pkg/release/release_amd64.go#L28

to version v0.14.0 instead of + Base ? 

I tagged that version in our quay repo, it's really an image that won't change with the openshift release, and installs the cni binary in /opt/cni/bin during the init phase of the flannel container.

**Comment by sallyom** [comment]
hm, weird, new error - maybe the test isn't waiting long enough? :eyes:

**Comment by sallyom** [comment]
I believe increasing the kuttl timeout will help here :)

**Comment by sallyom** [comment]
@fzdarsky look! here's a branch rebasing to `4.11.nightly` - see? ezpz https://github.com/openshift/microshift/pull/689 :)

**Comment by sallyom** [comment]
there are a few PRs going through in CI repo that hopefully will help here
* https://github.com/openshift/release/pull/28694
* https://github.com/openshift/release/pull/28355

**Comment by sallyom** [comment]
@mangelajo @fzdarsky @copejon I know what the issue is! The service-ca, node-resolver, dns images are failing to be pulled, because the pull secret/registry auth is missing! These component images are private

**Comment by sallyom** [comment]
@copejon @fzdarsky 
For development, with OCP (and CI/unreleased/private images) users will need to add a pull-secret file to /etc/crio/crio.conf.d - I have a PR for this in openshift/release, once that merges we can remove the commit that adds it here. I wanted to prove that this was the issue.

File: pkg/release/release_amd64.go [hunk 1]
```diff
@@ -17,18 +17,18 @@ limitations under the License.
...
+		"haproxy_router":                "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:a18d11de075fc9b11409f4bc4864bc73023d1510b0bb62b78f156d9a8819386e",
 		"kube_flannel":                  "quay.io/coreos/flannel:v0.14.0",
-		"kube_flannel_cni":              "quay.io/microshift/flannel-cni:" + Base,
-		"kube_rbac_proxy":               "quay.io/openshift/okd-content@sha256:baedb268ac66456018fb30af395bb3d69af5fff3252ff5d549f0231b1ebb6901",
+		"kube_flannel_cni":              "quay.io/microshift/flannel-cni:v0.14.0",
+		"kube_rbac_proxy":               "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:08e8b4004edaeeb125ced09ab2c4cd6d690afaf3a86309c91a994dec8e3ccbf3",
 		"kubevirt_hostpath_provisioner": "quay.io/kubevirt/hostpath-provisioner:v0.8.0",
 		"pause":                         "k8s.gcr.io/pause:3.6",
-		"service_ca_operator":           "quay.io/openshift/okd-content@sha256:0692de34bfd9f40455a5afc69fb47be56b1f386c55183b569d9d71d2969ff2e6",
+		"service_ca_operator":           "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:794adb97803879fc56331519893da0d8d17f31e058c0bca4972bc9937a253ef4",
```
Comment (sallyom) [inline pkg/release/release_amd64.go:32]: these images aren't public - should they be?

File: pkg/release/release_amd64.go [hunk 2]
```diff
@@ -17,18 +17,18 @@ limitations under the License.
...
+		"haproxy_router":                "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:608a8c9f893b3fae358028b44238f3184af2f3bd5cde844b8b2688583b8468ea",
 		"kube_flannel":                  "quay.io/coreos/flannel:v0.14.0",
-		"kube_flannel_cni":              "quay.io/microshift/flannel-cni:" + Base,
-		"kube_rbac_proxy":               "quay.io/openshift/okd-content@sha256:baedb268ac66456018fb30af395bb3d69af5fff3252ff5d549f0231b1ebb6901",
+		"kube_flannel_cni":              "quay.io/microshift/flannel-cni:v0.14.0",
+		"kube_rbac_proxy":               "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:08e8b4004edaeeb125ced09ab2c4cd6d690afaf3a86309c91a994dec8e3ccbf3",
 		"kubevirt_hostpath_provisioner": "quay.io/kubevirt/hostpath-provisioner:v0.8.0",
 		"pause":                         "k8s.gcr.io/pause:3.6",
-		"service_ca_operator":           "quay.io/openshift/okd-content@sha256:0692de34bfd9f40455a5afc69fb47be56b1f386c55183b569d9d71d2969ff2e6",
+		"service_ca_operator":           "quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:94a4cd55cb067f6d3747843b5532114806b6c81e8421e370bfe92ccafcc6bca7",
```
Comment (sallyom) [inline pkg/release/release_amd64.go:32]: these images aren't public - should they be? Does microshift pull these when starting from rpm? If so, why not the `quay.io/microshift/*:4.10.15` instead?

Comment (sallyom) [reply pkg/release/release_amd64.go:32]: added pull-secret - resolving this

File: packaging/images/microshift-aio/Dockerfile [hunk 3]
```diff
@@ -46,8 +46,7 @@ COPY $BUILD_PATH/unit /usr/lib/systemd/system/microshift.service
 COPY $BUILD_PATH/kubelet-cgroups.conf /etc/systemd/system.conf.d/kubelet-cgroups.conf
 COPY packaging/crio.conf.d/microshift.conf /etc/crio/crio.conf.d/microshift.conf
 
-# OCP_VERSION pushed ahead to 4.9.11 because aarch64 is now available, and it is backwards compatible
-RUN export OCP_VERSION=4.10.13 && \
+RUN export OCP_VERSION=4.10.15 && \
```
Comment (oglok) [inline packaging/images/microshift-aio/Dockerfile:49]: I think this should be 4.10.16, or is there a reason to keep it to .15?

Comment (sallyom) [reply packaging/images/microshift-aio/Dockerfile]: updated to 4.10.16

**Pull request merged**